invenio_rdm_records/stats/templates/aggregations/aggr_file_download/os-v1/aggr-file-download-v1.0.0.json

```diff
@@ -0,0 +1,69 @@
+{
+  "index_patterns": ["__SEARCH_INDEX_PREFIX__stats-file-download-*"],
+  "settings": {
+    "index": {
+      "refresh_interval": "1m"
+    }
+  },
+  "mappings": {
+    "dynamic_templates": [
+      {
+        "date_fields": {
+          "match_mapping_type": "date",
+          "mapping": {
+            "type": "date",
+            "format": "date_optional_time"
+          }
+        }
+      }
+    ],
+    "date_detection": false,
+    "dynamic": false,
+    "numeric_detection": false,
+    "properties": {
+      "timestamp": {
+        "type": "date",
+        "format": "date_optional_time"
+      },
+      "count": {
+        "type": "integer"
+      },
+      "unique_count": {
+        "type": "integer"
```

Comment:
will we ever max out the integer?

Reply:
In the long run, definitely – if we get to experience it in our life times however is a different question :smile: 
Zenodo is [using integers as well](https://github.com/zenodo/zenodo/blob/master/zenodo/modules/stats/templates/aggregations/v7/aggr-record-download-v1.0.0.json#L29), so I suspect the chances are relatively low.

The [official recommendations](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html#_which_type_should_i_use) on which integer type to pick are relatively vague.
That all being said, I wouldn't mind upgrading to `long`.
It seems like it wouldn't make a difference for storage anyway, and I doubt that this would become a performance bottleneck for us.

Reply:
`integer` supports up to 2^32 (so ~4 billion), which I think is more than fine (the most viewed record we have right now on Zenodo is at ~1.5 million).